src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/CreateLegalEntity/CreateLegalEntityCommandHandler.cs

```diff
@@ -55,27 +59,24 @@ public class CreateLegalEntityCommandHandler : IAsyncRequestHandler<CreateLegalE
                 owner.AccountId,
                 message.LegalEntity);
 
+            
+
             await CreateAuditEntries(owner, agreementView);
 
             await NotifyLegalEntityCreated(message.HashedAccountId, agreementView.LegalEntityId);
 
-            await PublishAgreementCreatedMessage(owner.AccountId, agreementView.LegalEntityId,
-                agreementView.Id);
+            await PublishAgreementCreatedMessage(_hashingService.DecodeValue(message.HashedAccountId),

```

Comment:
can we move the hashing service call outside of the method call so that it can be easily debugged later on?

Reply:
OK